core/deployment/src/main/java/io/quarkus/deployment/recording/PropertyUtils.java

```diff
@@ -22,9 +22,32 @@ public Property[] apply(Class<?> type) {
             List<Property> ret = new ArrayList<>();
             Method[] methods = type.getMethods();
 
+            List<Method> checkedMethodList = new ArrayList<>();
+
+            for (Method method : methods) {
+                if (method.getName().startsWith("get")) {
```

Comment:
I don't think this is enough of a test. If you have `getTest(String)` and `isTest()`, you will declare a duplicate while actually there is not and this is properly tested in the loop below.

I think you should extract `i.getName().startsWith("get") && i.getName().length() > 3 && i.getParameterCount() == 0 && i.getReturnType() != void.class` and similar to some `isGetGetter()`/`isIsGetter()`/`isSetter()` methods and use them in both loops.

Reply:
Oh and thanks for given it a try :).